### PR TITLE
Split SMTCheckerTests in CI

### PR DESCRIPTION
**Pull request by leonardoalt**

### CODE REVIEW

File: .circleci/soltest_all.sh [hunk 1]
```diff
@@ -57,7 +57,7 @@ echo "Running steps $RUN_STEPS..."
 
 STEP=1
 
-[[ " $RUN_STEPS " =~ " $STEP " ]] && EVM=istanbul OPTIMIZE=1 ABI_ENCODER_V2=1 "${REPODIR}/.circleci/soltest.sh"
+[[ " $RUN_STEPS " =~ " $STEP " ]] && EVM=istanbul OPTIMIZE=1 ABI_ENCODER_V2=1 SOLTEST_ARGS="--no-smt" "${REPODIR}/.circleci/soltest.sh"
```
Comment (axic) [inline .circleci/soltest_all.sh]: Perhaps it is time to introduce LATEST_EVM on top?

Comment (chriseth) [reply .circleci/soltest_all.sh]: This also disables the smt optimizer tests, doesn't it? I would feel safer by using the `-t` option with negated patterns, if that is possible. Or introduce another `@`-tag.

Comment (leonardoalt) [reply .circleci/soltest_all.sh]: Different PR? :p

Comment (leonardoalt) [reply .circleci/soltest_all.sh]: @chriseth ok!

**Comment by leonardoalt** [comment]
Maybe we don't need to do this actually, because of https://github.com/ethereum/solidity/pull/9873

**Comment by chriseth** [comment]
I think this might still make sense, but you have to adjust the code in line 37+ - the circelci logic does not check how long each job takes, so you have to manually distribute the larger ones to their own executors.

**Comment by leonardoalt** [comment]
Right. I moved the run with only SMTChecker tests to the beginning, so it's the first - and heaviest expected - run.

**Review by chriseth — APPROVED** [review]

**Pull request merged**